internal/webhooks/agent/pod_defaulter.go

```diff
@@ -163,6 +164,41 @@ func (r *podMutator) Default(ctx context.Context, obj runtime.Object) error {
 		},
 	})
 
+	if metav1.HasLabel(pod.ObjectMeta, constants.AgentLabelSmartTriggersConfigMaps) {
+		// Mount the Smart Triggers colume
+		readOnlyMode := int32(0440)
+		smartTriggersConfigMapNames := getSmartTriggersConfigMapNames(pod.Labels)
+		for _, triggerMap := range smartTriggersConfigMapNames {
+			pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
+				Name: "trigger-" + triggerMap,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: triggerMap,
+						},
+						DefaultMode: &readOnlyMode,
+					},
+				},
+			})
+		}
+
+		// Mount the triggers specified in the pod labels under /tmp/smart-triggers
+		for _, triggerMap := range getSmartTriggersConfigMapNames(pod.Labels) {
+			container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+				Name:      "trigger-" + triggerMap,
+				MountPath: "/tmp/smart-triggers",
```

Comment:
I think this path should be configurable too, just to be safe. It's all unlikely but maybe there is some conflict with something else at this path, or maybe `/tmp` doesn't exist or isn't writable. Best to make sure there's a way to configure around these potential issues.

Reply:
Maybe we could mount it under the empty dir volume we create? We'd just have to account for the file size of things we put there and our resource limits we defined for the volume.